r2r/vector_dbs/lancedb/base.py

```diff
@@ -0,0 +1,147 @@
+import logging
+import os
+from typing import Optional, Union
+
+from r2r.core import VectorDBProvider, VectorEntry, VectorSearchResult
+
+logger = logging.getLogger(__name__)
+
+
+class LanceDB(VectorDBProvider):
+    def __init__(
+        self, provider: str = "lancedb", db_path: Optional[str] = None
+    ) -> None:
+        logger.info("Initializing `LanceDB` to store and retrieve embeddings.")
+
+        super().__init__(provider)
+
+        if provider != "lancedb":
+            raise ValueError(
+                "LanceDB must be initialized with provider `lancedb`."
+            )
+
+        try:
+            import lancedb
+        except ImportError:
+            raise ValueError(
+                f"Error, `lancedb` is not installed. Please install it using `pip install lancedb`."
+            )
+
+        self.db_path = db_path
+        try:
+            self.client = lancedb.connect(uri=self.db_path or os.environ.get("LANCEDB_URI"), api_key=os.environ.get("LANCEDB_API_KEY") or None, region=os.environ.get("LANCEDB_REGION") or None)
+        except Exception as e:
+            raise ValueError(
+                f"Error {e} occurred while attempting to connect to the lancedb provider."
+            )
+        self.collection_name: Optional[str] = None
+
+    def initialize_collection(
+        self, collection_name: str, dimension: int
+    ) -> None:
+        self.collection_name = collection_name
+
+        try:
+            import pyarrow
+        except ImportError:
+            raise ValueError(
+                f"Error, `pyarrow` is not installed. Please install it using `pip install pyarrow`."
+            )
+
+        table_schema = pyarrow.schema(
+            [
+                pyarrow.field("id", pyarrow.string()),
+                pyarrow.field(
+                    "vector", pyarrow.list_(pyarrow.float32(), dimension)
+                ),
+                # TODO Handle storing metadata
```

Comment:
@AyushExel I need a column in the table to store metadata information associated  with the vector. 

Metadata has the following type: 
```python
MetadataValues = Union[str, int, float, bool, List[str]]
Metadata = Dict[str, MetadataValues]
```

How can I achieve this with Pyarrow?